### PR TITLE
Rename tar_<filetype>() to tar_<pkg>_<data type>()

### PR DESCRIPTION
**Pull request by Aariq**
This PR renames `tar_geotiff()` to `tar_terra_rast()`.  It separates out the format into a new function, `format_terra_rast()` that returns the output of a `tar_format()` call. ~~I separated it out so it could accept a `filetype` argument where users can choose a GDAL driver (from a list of possibilities we have implemented and tested with `targets`).~~

I'm thinking maybe we'll end up having tar_ functions like `tar_<pkg name>_<data type/class>()` with some default storage formats rather than naming the functions after the file type.  What do you think?

Somewhat addresses #4

### CODE REVIEW

File: R/tar-terra.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#' @export
+tar_terra_rast <- function(name,
+                           command,
+                           pattern = NULL,
+                           packages = targets::tar_option_get("packages"),
+                           tidy_eval = targets::tar_option_get("tidy_eval"),
+                           library = targets::tar_option_get("library"),
+                           format = format_terra_rast(),
```
Comment (Aariq) [inline R/tar-terra.R:8]: This here is the main difference (sorry, looks like GitHub is registering this as an entirely  new file since I renamed it)

Comment (njtierney) [reply R/tar-terra.R:8]: Looks good to me!

**Comment by njtierney** [comment]
I really like the idea of having  `tar_<pkg name>_<data type/class>()`. I think this will help encourage discoverability for users. Nice one!

File: R/formats.R [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    } else if (filetype == "netCDF") {
+        write_fun <- function(object, path) {
+            terra::writeRaster(
+                x = object,
+                filename = path,
+                overwrite = TRUE,
+                filetype = "netCDF"
+            )
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Aariq) [reply R/formats.R:22]: This is excellent, but I'm going to move it to an issue because I think there are API things to discuss before implementing this.

**Review by brownag — APPROVED** [review]
This looks good to me!

**Pull request merged**